firewood/src/merkle/node/leaf.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+use std::fmt::{Debug, Error as FmtError, Formatter};
+
+use bincode::Options;
+
+use super::{Data, Encoded};
+use crate::merkle::{from_nibbles, PartialPath};
+
+#[derive(PartialEq, Eq, Clone)]
+pub struct LeafNode {
+    pub(crate) path: PartialPath,
+    pub(crate) data: Data,
+}
+
+impl Debug for LeafNode {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), FmtError> {
+        write!(f, "[Leaf {:?} {}]", self.path, hex::encode(&*self.data))
+    }
+}
+
+impl LeafNode {
+    pub fn new<P: Into<PartialPath>, D: Into<Data>>(path: P, data: D) -> Self {
+        Self {
+            path: path.into(),
+            data: data.into(),
+        }
+    }
+
+    pub fn path(&self) -> &PartialPath {
```

Comment:
nit: since these are `pub fn` we should reduce the visibility of the members themselves, or get rid of this.

Reply:
Agreed (both `path` and `data` methods are redundant

https://github.com/ava-labs/firewood/issues/364